### PR TITLE
Fix clippy for latest stable toolchain

### PR DESCRIPTION
**Pull request by octol**
# Description

- Unpin Rust toolchain for wallet
- cargo clippy --fix
- clippy::redundant-guards
- clippy::incorrect_partial_ord_impl_on_ord_type
- clippy::unwrap-or-default
- noop_method_call

### CODE REVIEW

File: common/nymsphinx/params/src/packet_sizes.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ pub enum PacketSize {
 impl PartialOrd for PacketSize {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         // order them by actual packet size
-        self.size().partial_cmp(&other.size())
+        Some(self.cmp(other))
```
Comment (jstuczyn) [inline common/nymsphinx/params/src/packet_sizes.rs:92]: Are we sure this is doing the same thing as before? We seem to no longer be comparing on `size()`

Comment (jstuczyn) [reply common/nymsphinx/params/src/packet_sizes.rs:92]: Ah. `Ord` already does it.

File: ephemera/src/core/ephemera.rs [hunk 2]
```diff
@@ -120,7 +120,7 @@ impl<A: Application> Ephemera<A> {
         while !shutdown.is_shutdown() {
             tokio::select! {
                 biased;
-                _ = shutdown.recv() => {
+                () = shutdown.recv() => {
```
Comment (jstuczyn) [inline ephemera/src/core/ephemera.rs:123]: I'm surprised clippy cares about that one. I guess you'll have plenty of those to fix now

**Review by jstuczyn — APPROVED** [review]

File: ephemera/src/core/ephemera.rs (hunk 2 above)
Comment (octol) [reply ephemera/src/core/ephemera.rs:123]: We're seeing it only in ephemera as it has pedantic enabled

**Pull request merged**